contentcuration/contentcuration/static/js/edit_channel/preview/views.js

```diff
@@ -83,44 +83,7 @@ var PreviewView = BaseViews.BaseView.extend({
     },
     generate_preview:function(force_load){
         if(this.current_preview){
-            extension = this.current_preview.file_format;
-            var preview_template;
-            switch (extension){
-                case "png":
-                case "jpg":
-                case "jpeg":
-                    preview_template = require("./hbtemplates/preview_templates/image.handlebars");
-                    break;
-                case "pdf":
-                case "PDF":
-                case "vtt":
-                case "srt":
-                    preview_template = require("./hbtemplates/preview_templates/document.handlebars");
-                    break;
-                case "mp3":
-                    preview_template = require("./hbtemplates/preview_templates/audio.handlebars");
-                    break;
-                case "mp4":
-                    preview_template = require("./hbtemplates/preview_templates/video.handlebars");
-                    break;
-                case "perseus":
-                    preview_template = require("./hbtemplates/preview_templates/exercise.handlebars");
-                    break;
-                case "zip":
-                    preview_template = require("./hbtemplates/preview_templates/html5.handlebars");
-                    break;
-                default:
-                    preview_template = require("./hbtemplates/preview_templates/default.handlebars");
-            }
-            this.$("#preview_window").html(preview_template({
-                source: this.current_preview.storage_url,
-                extension:this.current_preview.mimetype,
-                checksum:this.current_preview.checksum,
-                subtitles : this.get_subtitles()
-            }));
-            if(force_load && this.model.get('kind') === "video"){
-                $("#preview_window video").load();
-            }
+            _.defer(render_preview, this.$("#preview_window"), this.current_preview, this.get_subtitles(), force_load && this.model.get('kind') === "video");
```

Comment:
render_preview is now a separate method so we can support rendering individual content previews at other places. _.defer ensures that the DOM tree has been fully rendered before trying to inject a preview